survive-the-monsters.js

```diff
@@ -80,15 +86,81 @@ function normalizeInput() {
     input[0] /= m;
     input[1] /= m;
 }
-function applyMovement(dt) {
-    const speed = movementConfig.speed * dt;
+
+function applyPlayerMovement(dt) {
+    const speed = movementConfig.playerSpeed * dt;
     playerX += input[0] * speed;
     playerY += input[1] * speed;
 }
+function applyEnemiesMovement(dt) {
+    const speed = movementConfig.enemiesSpeed * dt;
+    
+    for (let enemyId = 0; enemyId <= enemiesLargestId; enemyId++) {
+        if (!enemiesState[enemyId]) {
+            continue;
+        }
+        const manDistX = playerX - enemiesX[enemyId];
+        const manDistY = playerY - enemiesY[enemyId];
+        const eucDistSq = manDistX**2 + manDistY**2;
+
+        let eucDist;
+        let eigX;
+        let eigY;
+        if (eucDistSq < 1) {
+            eucDist = 1;
+            eigX = manDistX;
+            eigY = manDistY;
+        } else {
+            eucDist = eucDistSq**0.5;
+            eigX = manDistX / eucDist;
+            eigY = manDistY / eucDist;
+        }
+
+        enemiesX[enemyId] += eigX * speed;
+        enemiesY[enemyId] += eigY * speed;
+    }
+}
+
 function updatePositions() {
-    for (let enemyId = 0; enemyId < ENEMIES_COUNT; enemyId++) {
+    for (let enemyId = 0; enemyId <= enemiesLargestId; enemyId++) {
+        if (!enemiesState[enemyId]) {
+            continue;
+        }
         const x = sceneCenterX - playerX + enemiesX[enemyId];
```

Comment:
**praise**: Great use of `setInterval` to spawn enemies at regular intervals, which aligns well with the task requirements.